accepted/2025/direct-tool-execution.md

```diff
@@ -0,0 +1,114 @@
+# dotnet tool exec and dnx
+
+**Owner**: [Marc Paine](https://github.com/marcpopMSFT)
+
+## `dotnet tool exec` Summary
+
+This proposal introduces a new .NET CLI command, `dotnet tool exec`, which enables users to launch a .NET tool directly by specifying its package ID. The command will ensure the tool is installed (downloading the latest version to the NuGet cache if necessary) and then execute it with the provided arguments. Optionally, users can specify a particular version of the tool to run.
+
+This feature streamlines the process of running tools, removing the need for explicit installation or manifest management for ad-hoc tool usage.
+
+## Motivation
+
+Currently, running a .NET tool requires a two-step process: installing the tool (either globally or locally) and then invoking it. This is cumbersome for scenarios where a tool is only needed temporarily or for a single use. By providing a single command to install (if needed) and execute a tool, we improve developer productivity and lower the barrier to using .NET tools.
+
+This also enables scenarios such as:
+
+- Running a tool in CI/CD pipelines without polluting the global or local tool manifest.
+- Quickly trying out tools without a permanent install.
+- Scripting and automation where ephemeral tool usage is desired.
+
+## CLI Parameters
+
+The `dotnet tool exec` command will support the following parameters:
+
+| Parameter                | Description                                                                                  |
+|--------------------------|----------------------------------------------------------------------------------------------|
+| `<package-id>`           | The NuGet package ID of the tool to execute.                                                 |
+| `--version <version>`    | (Optional) The specific version of the tool to run. If not specified, the latest is used.    |
```

Comment:
While we should support `--version`, I think the preferred way to specify the version should be via the package@version syntax, for example `dotnet tool exec dotnetsay@1.2.3 -- Hello, World!`.

Reply:
I'm a bit surprised by this. Do we use the `@` syntax much in .NET? The `dotnet package add` syntax uses `--version` so it felt natural to me. I disagree per se, just have an initial question on whether it would be inconsistent.

Reply:
We've started adding support for `id@version` syntax, first I think in workloads and now in some tool and NuGet commands.  We think it's a nicer syntax and want to support it generally. /cc @baronfel